core/protocol/all_services.go

```diff
@@ -574,6 +570,15 @@ func (svcs *allServices) setupNetParameters(powWatchers []netparams.WatchParam)
 				return svcs.eventForwarderEngine.SetupEthereumEngine(svcs.ethClient, svcs.eventForwarder, svcs.conf.EvtForward.Ethereum, ethCfg, svcs.assets)
 			},
 		},
+		{
+			Param: netparams.EthereumOraclesEnabled,
+			Watcher: func(ctx context.Context, isEnabled int64) error {
```

Comment:
Is this so we don't run a poller thread when no enable ethereum oracles?  IMO it seems preferable just to always run it and remove the additional locking, if there are no specs if will not call out.  But maybe I have misunderstood?

Reply:
well yes, that was my thought; no chance if it being able to mess anything up somehow if it's not running - but I do take your point.